pkg/controllers/hpareplicassyncer/hpa_replicas_syncer_controller.go

```diff
@@ -125,8 +126,14 @@ func (r *HPAReplicasSyncer) updateScaleIfNeed(ctx context.Context, workloadGR sc
 
 		return nil
 	}
-
-	if scale.Spec.Replicas != hpa.Status.DesiredReplicas {
+	// DesiredReplicas only has meaning when ScalingActive and AbleToScale are both True. At any other time, it is meaningless.
+	// So the workload does not need to be updated.
+	// Refer to:
+	// https://github.com/kubernetes/kubernetes/blob/v1.28.0/pkg/controller/podautoscaler/horizontal.go#L797
+	// https://github.com/kubernetes/kubernetes/blob/v1.28.0/pkg/controller/podautoscaler/horizontal.go#L343
+	if scale.Spec.Replicas != hpa.Status.DesiredReplicas &&
+		findConditionTrue(hpa.Status, autoscalingv2.ScalingActive) &&
+		findConditionTrue(hpa.Status, autoscalingv2.AbleToScale) {
```

Comment:
Does it mean that as long as the state of the HPA in one cluster is not normal, the replica information in the other clusters will not be collected correctly either? Wouldn't this setup be too strong?

Reply:
> Does it mean that as long as the state of the HPA in one cluster is not normal, the replica information in the other clusters will not be collected correctly either?

Yes. do you have any suggestions?

Reply:
I have an idea that could be incorrect. During state aggregation, if the status from a cluster is not working properly, add the CurrentReplicas in the status to the DesiredReplicas

Reply:
> add the CurrentReplicas in the status to the DesiredReplicas


🤔，hpa  is not work, so both CurrentReplicas and DesiredReplicas are meaningless.
 They cannot reflect the true situation of the workload anymore.

Reply:
Is it possible to consider handling the collection of work's status by not updating the work status if the hpa status is not normal, so that it stays in its previous normal status?

Reply:
> Is it possible to consider handling the collection of work's status by not updating the work status if the hpa status is not normal, so that it stays in its previous normal status?

I don't quite understand, what are the benefits of doing this? 🤔

Reply:
Because I have a feeling that the current solution is not very friendly to the situation where a cluster hpa is not working properly, and there is a possibility that it will stay not working properly, which will lead to other clusters not being able to collect the information correctly